pkg/cli/advisory_discover.go

```diff
@@ -1,40 +1,51 @@
 package cli
 
 import (
+	"fmt"
+	"log"
 	"net/http"
+	"os"
+	"time"
 
 	"github.com/spf13/cobra"
 	"github.com/wolfi-dev/wolfictl/pkg/advisory"
-	"github.com/wolfi-dev/wolfictl/pkg/vuln/sftracker"
+	"github.com/wolfi-dev/wolfictl/pkg/vuln/nvdapi"
 )
 
 const defaultSecfixesTrackerHostname = "secfixes-tracker-q67u43ydxq-uc.a.run.app"
 
+//nolint:gosec // This is not a hard-coded credential value, it's the name of the env var to reference.
+const envVarNameForNVDAPIKey = "WOLFICTL_NVD_API_KEY"
+
 func AdvisoryDiscover() *cobra.Command {
 	p := &discoverParams{}
 	cmd := &cobra.Command{
 		Use:           "discover",
 		Short:         "search for new potential vulnerabilities and create advisories for them",
 		SilenceErrors: true,
 		RunE: func(cmd *cobra.Command, args []string) error {
+			start := time.Now()
+
 			index, err := newConfigIndexFromArgs(args...)
 			if err != nil {
 				return err
 			}
 
-			secfixesTracker := sftracker.New(
-				p.secfixesTrackerHostname,
-				http.DefaultClient,
-			)
+			apiKey := p.resolveNVDAPIKey()
 
 			err = advisory.Discover(advisory.DiscoverOptions{
-				Index:                 index,
-				VulnerabilitySearcher: secfixesTracker,
+				Configs:               index,
+				PackageRepositoryURL:  p.packageRepositoryURL,
+				Arch:                  "x86_64",
```

Comment:
Are there any weird implications of hard-coding this here? If a package ever exists that's only built for arm64 (unlikely, sure) will it never have vulns discovered?

Reply:
No yeah, this is a good callout. Ideally the `discover` command would consult apkindexes for all of our supported architectures, and then use the union set in its logic here.

Is there an authoritative place or way to determine the current set of apkindexes that exist for the distro at a given time?

Reply:
I don't believe there is. cc @kaniini 

We can invent one, like `archs.json` or something.

Reply:
Okay cool. For now, I'll change this to be a slice of strings instead of a string, and pass in both `x86_64` and `aarch64` — so that we don't miss anything in the meantime. And then, if it's good with you, we can update to refer to that new authority in the future when it's available.